spec/system/exercises_system_spec.rb

```diff
@@ -0,0 +1,162 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe 'Exercise creation', :js do
+  let!(:ruby) { create(:ruby) }
+
+  before do
+    visit(sign_in_path)
+    fill_in('email', with: teacher.email)
+    fill_in('password', with: attributes_for(:teacher)[:password])
+    click_button(I18n.t('sessions.new.link'))
+    wait_for_ajax
+  end
+
+  context 'when an exercise created' do
+    let(:teacher) { create(:teacher) }
+    let(:submission_deadline) { 3.months.from_now.beginning_of_minute }
+    let(:late_submission_deadline) { submission_deadline + 1.week }
+    let(:title) { 'Ruby challenge' }
+    let(:internal_title) { 'Project Ruby: Table flip' }
+
+    let(:code) do
+      <<~RUBY
+        def self.❨╯°□°❩╯︵┻━┻
+          puts "Calm down, bro"
+        end
+      RUBY
+    end
+
+    let(:description) do
+      <<~TEXT
+        Ruby challenge
+
+        Do something with Ruby.
+      TEXT
+    end
+
+    before do
+      visit(exercises_path)
+      click_on I18n.t('shared.new_model', model: Exercise.model_name.human)
+      wait_for_ajax
+    end
+
+    it 'creates an exercise with nested data' do
+      fill_in Exercise.human_attribute_name(:title), with: title
+      fill_in Exercise.human_attribute_name(:internal_title), with: internal_title
+
+      # description
+      within('.markdown-editor__wrapper') do
+        find('.ProseMirror').set(description)
+      end
```

Comment:
I wonder if these very specific class selectors might cause brittleness in the future if a refactoring is upcoming